core/common/src/main/java/alluxio/ConstantPropertyKey.java

```diff
@@ -0,0 +1,510 @@
+    public static final String USER_UFS_FILE_READER_CLASS = "alluxio.user.ufs.file.reader.class";
+    public static final String USER_UFS_FILE_WRITER_CLASS = "alluxio.user.ufs.file.writer.class";
+    public static final String USER_UFS_BLOCK_READ_LOCATION_POLICY =
+        "alluxio.user.ufs.block.read.location.policy";
+    public static final String USER_UFS_BLOCK_READ_LOCATION_POLICY_DETERMINISTIC_HASH_SHARDS =
+        "alluxio.user.ufs.block.read.location.policy.deterministic.hash.shards";
+    public static final String USER_UFS_BLOCK_READ_CONCURRENCY_MAX =
+        "alluxio.user.ufs.block.read.concurrency.max";
+    public static final String USER_UFS_BLOCK_OPEN_TIMEOUT_MS =
+        "alluxio.user.ufs.block.open.timeout.ms";
+    public static final String USER_SHORT_CIRCUIT_ENABLED = "alluxio.user.short.circuit.disabled";
+
+    //
+    // FUSE integration related properties
+    //
+    public static final String FUSE_CACHED_PATHS_MAX = "alluxio.fuse.cached.paths.max";
+    public static final String FUSE_DEBUG_ENABLED = "alluxio.fuse.debug.enabled";
+    public static final String FUSE_FS_NAME = "alluxio.fuse.fs.name";
+    public static final String FUSE_FS_ROOT = "alluxio.fuse.fs.root";
+    public static final String FUSE_MAXWRITE_BYTES = "alluxio.fuse.maxwrite.bytes";
+    public static final String FUSE_MOUNT_DEFAULT = "alluxio.fuse.mount.default";
+
+    //
+    // Security related properties
+    //
+    public static final String SECURITY_AUTHENTICATION_CUSTOM_PROVIDER_CLASS =
+        "alluxio.security.authentication.custom.provider.class";
+    public static final String SECURITY_AUTHENTICATION_SOCKET_TIMEOUT_MS =
+        "alluxio.security.authentication.socket.timeout.ms";
+    public static final String SECURITY_AUTHENTICATION_TYPE =
+        "alluxio.security.authentication.type";
+    public static final String SECURITY_AUTHORIZATION_PERMISSION_ENABLED =
+        "alluxio.security.authorization.permission.enabled";
+    public static final String SECURITY_AUTHORIZATION_PERMISSION_SUPERGROUP =
+        "alluxio.security.authorization.permission.supergroup";
+    public static final String SECURITY_AUTHORIZATION_PERMISSION_UMASK =
+        "alluxio.security.authorization.permission.umask";
+    public static final String SECURITY_GROUP_MAPPING_CACHE_TIMEOUT_MS =
+        "alluxio.security.group.mapping.cache.timeout.ms";
+    public static final String SECURITY_GROUP_MAPPING_CLASS =
+        "alluxio.security.group.mapping.class";
+    public static final String SECURITY_LOGIN_USERNAME = "alluxio.security.login.username";
+
+    private Name() {
+    } // prevent instantiation
+  }
+
+  /** A map from default property key's string name to the key. */
+  static final Map<String, ConstantPropertyKey> DEFAULT_KEYS_MAP = new HashMap<>();
+
+  /**
+   * @param keyStr string of property key
+   * @return whether the input is a valid property name
+   */
+  public static boolean isValid(String keyStr) {
```

Comment:
do the static methods in this class need to be public or could they be package private?